lib/modules/datasource/docker/index.ts

```diff
@@ -812,7 +812,7 @@ export class DockerDatasource extends Datasource {
   @cache({
     namespace: 'datasource-docker-digest',
     key: (
-      { registryUrl, packageName }: GetReleasesConfig,
+      { registryUrl, packageName, currentValue, currentDigest }: DigestConfig,
```

Comment:
I think you need to use value and digerst in cache key, otherwise we get into trouble with caching

Reply:
You're right, I was able to verify this with breakpoints on `finalKey` and `finalNamespace` in the cache function.

**Until now**
- The cache key is always in the form: `https://index.docker.io:hashicorp/vault-k8s:0.12.0` as composed here: https://github.com/renovatebot/renovate/blob/49170603871b8a1bec946cd3682c0b83b34cf6fe/lib/modules/datasource/docker/index.ts#L823

- `getDigest()` will always return the same digest, no matter if the user provides a manifest list or an architecture digest in `currentDigest`. If no `currentDigest` is provided, it could return either, depending on what is currently cached.

**What do we want?**

"_For the new version `newTag` give me the digest based on `currentDigest`, if available_"

I've now added `currentDigest` to the key to make caching deterministic. This means:
- With no `currentDigest` provided (= no digest pinned) -> cache new manifest digest
- With `currentDigest` (architecture or manifest list digest) -> cache new architecture or new manifest list digest

Tested successfully with https://github.com/Churro/renovate-archdigests and other repos, where the cache was hit successfully when running renovate multiple times.